### ISSUE TITLE
Drop function's rest support

### ISSUE THREAD

**Issue by Constellation** [estools/escope#53]
Use `RestElement` and drop function's rest  support.

/cc @nzakas

**Comment by nzakas** [comment]
:+1: we also will make this switch as a breaking change

**Comment by Constellation** [comment]
@nzakas 

Could you tell me when this is switched?
At that time, I'll drop `function.rest` and handle `RestElement` in function's `params`.

**Comment by nzakas** [comment]
Yup. We have this issue tracking it: https://github.com/eslint/espree/issues/84

Is this blocking for the module-related changes?
